### PR TITLE
Bingo elastic python

### PR DESCRIPTION
**Pull request by khyurri**
What should be done:

- [x] Autotests using Actions
- [x] Readme with usage examples
- [x] Build and deployment documentation

### CODE REVIEW

File: api/plugins/bingo-elastic/python/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+*CAVEAT*: Elasticsearch doesn't have strict notion of commit, so records might appear in the index later on
+Read more about it here -  https://www.elastic.co/guide/en/elasticsearch/reference/master/index-modules.html#index-refresh-interval-setting
+
+For indexing one record the the method `ElasticRepository.index_record` can be used 
+
+#### Retrieve similar records from Elasticsearch
+
+```
+from bingo_elastic.predicates import TanimotoSimilarityMatch
+alg = TanimotoSimilarityMatch(target, 0.9)
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/python/README.md:111]: let's specify just SimilarityMatch

File: api/plugins/bingo-elastic/python/README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+similar_records = repository.filter(similarity=alg, limit=20)
+```
+
+In this case we requested top-20 most similar molecules compared to `target` based on Tanimoto similarity metric
+
+Supported similarity algorithms:
+- `TanimotoSimilarityMatch`
+- `EuclidSimilarityMatch`
+- `TverskySimilarityMatch`
+- `ExactMatch`
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/python/README.md:121]: Not a similarity, let's separate

File: api/plugins/bingo-elastic/python/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+#### Find exact records from Elasticsearch
+
+```
+alg = TanimotoSimilarityMatch(target)
+exact_records = repository.filter(similarity=alg, limit=20)
+```
+
+In this case we requested top-20 candidate molecules with exact same fingerprint to `target`.
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/python/README.md:130]: default still 20, not 10 like in Java API?

**Review by MysterionRise — CHANGES_REQUESTED** [review]

File: api/plugins/bingo-elastic/python/README.md (hunk 3 above)
Comment (khyurri) [reply api/plugins/bingo-elastic/python/README.md:130]: This limit explicitly set as argument `limit=20` 

I changed default limit to 10 in `ElasticRepository.filter` method

File: api/plugins/bingo-elastic/python/bingo_elastic/elastic.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        self,
+        *,
+        host: Union[str, list[str]] = "localhost",
+        port: int = 9200,
+        scheme: str = ""
+    ) -> None:
+        """
+        :param host: host or list of hosts
+        :param port:
+        :param scheme: http or https
```
[the remark this answers is not in the document]
Comment (khyurri) [reply api/plugins/bingo-elastic/python/bingo_elastic/elastic.py:30]: Added

File: api/plugins/bingo-elastic/python/README.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+`target` should be an instance of `IndigoRecord` class. 
+
+
+
+
+#### Subsctructure match of the records from Elasticsearch
+
+```
+exact_records = repository.filter(substructure=target)
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/python/README.md:137]: submatch_records

File: api/plugins/bingo-elastic/python/bingo_elastic/elastic.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            chunk_size=chunk_size,
+        ):
+            pass
+            # TODO: add post processing
+
+    def index_records_parallel(
+        self,
+        records: Generator,
+        chunk_size: int = 500,
+        thread_count: int = 4,
```
[the remark this answers is not in the document]
Comment (MysterionRise) [reply api/plugins/bingo-elastic/python/bingo_elastic/elastic.py:79]: or at least place it as TODO fixes?

**Review by MysterionRise — APPROVED** [review]
All good, ready to merge!

File: api/plugins/bingo-elastic/python/bingo_elastic/__init__.py [hunk 7]
```diff
@@ -0,0 +1 @@
+__version__ = "1.4.1"
```
Comment (mkviatkovskii) [inline api/plugins/bingo-elastic/python/bingo_elastic/__init__.py:1]: Do we need to set this manually, as well as in Poetry?

Comment (khyurri) [reply api/plugins/bingo-elastic/python/bingo_elastic/__init__.py:1]: It will be cool, could you add it to the CI pipeline?

**Pull request merged**